plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/IngestBootstrapper.java

```diff
@@ -126,6 +141,39 @@ boolean validClusterState(ClusterState state) {
         }
     }
 
+    boolean isIngestTemplateInstallationRequired(MetaData metaData) {
+        if (metaData.getTemplates().containsKey(INGEST_INDEX_TEMPLATE_NAME)) {
+            logger.trace("not installing ingest index template, because it already is installed");
+            return false;
+        }
+        return true;
+    }
+
+    void forkAndInstallIngestIndexTemplate() {
+        threadPool.executor(ThreadPool.Names.GENERIC).execute(() -> {
+            try {
+                installIngestIndexTemplate();
+            } catch (IOException e) {
+                logger.error("Failed to install .ingest index template", e);
```

Comment:
so, if we fail to install the template, we just flag and continue working as usual? So ingest template is optimistic, and not a requirement.